lib/OpenQA/Schema/Result/Jobs.pm

```diff
@@ -470,7 +470,12 @@ sub to_hash {
     my ($job, %args) = @_;
     my $j = _hashref($job, qw(id name priority state result clone_id t_started t_finished group_id blocked_by_id));
     if ($j->{group_id}) {
-        $j->{group} = $job->group->name;
+        my $job_group = $job->group;
+        $j->{group} = $job_group->name;
+        if (my $parent_group = $job_group->parent) {
+            $j->{parent_group_id} = $parent_group->id;
+            $j->{parent_group}    = $parent_group->name;
+        }
```

Comment:
Considering that the `to_hash` function is called in several places which to not need the extra lookup it would be better make this optional like it is already done for incorporating asset info.

Reply:
How about `if ($args{parent})` and always set the values. If you're using that flag you're probably sure you want the fields to be there. And otherwise you can't tell the difference between unset and wrong API usage.

Reply:
The API might even return the parent group by default. My main concern is that the use `to_hash` also in places which have nothing to do with the API and it would make sense to save the extra SQL query here.

Reply:
Agreed, I kinda had the thought as well, but didn't mention it here since this isn't the API

Reply:
@Martchus I modified the code, please help to confirm